unity-environment/Assets/ML-Agents/Examples/Shared Assets/Scripts/RayPerception.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+
+public class RayPerception : MonoBehaviour
+{
+
+    List<float> perceptionVector = new List<float>();
+
+    public List<float> Percieve(float rayDistance,
+                         float[] rayAngles, string[] detectableObjects,
+                          float startOffset, float endOffset)
```

Comment:
Can these parameters be exposed at the gameObject level.
Feels weird to have to pass the same parameters when calling AddVectorObs

Reply:
No, because sometimes you want to use more than one set ray at a time. If they were component-level, would you need different components for different sets of rays?